### ISSUE TITLE
[FR] Support Cython

### ISSUE THREAD

**Issue by webknjaz** [PyCQA/pydocstyle#473]
Hi, it'd be great if this tool could enforce docstrings in Cython. Would it be hard to implement?

**Comment by samj1912** [comment]
Not sure - but it isn't straightfoward afaict - it would involve updating https://github.com/PyCQA/pydocstyle/blob/master/src/pydocstyle/parser.py to support cython. Currently we heavily rely on the tokenize module to parse files, we also use the ast module for complex use cases around google/numpy styles. Unless cython has something equivalent, it will require a bit of research and implementing/using some library that can parse cython source code per our needs.

**Comment by webknjaz** [comment]
Right... It'd probably require integrating something like https://github.com/cython/cython/tree/master/Cython/Plex
